### PR TITLE
Make it possible to run HermiT on EMMO

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
Make it possible to run HermiT on EMMO, by:
* Add support for additional keyword arguments that `sync_reasoner()` may pass to the `save()` method.
* Remove custom datatypes before running HermiT. Readd the removed data properties after HermiT completes.
* Add `--java-executable` and `--java-memory` arguments to ontoconvert to control how to run java.

## Type of change

- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.
- [ ] Test update.

## Checklist

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

File: ontopy/ontology.py [hunk 1]
```diff
@@ -940,7 +942,9 @@ def save(
                 raise TypeError("`filename` and `format` cannot both be None.")
         else:
             file = filename
-        filepath = os.path.join(dir, file)
+        filepath = os.path.join(
+            dir, file if isinstance(file, str) else file.name
+        )
```
Comment (francescalb) [inline ontopy/ontology.py:947]: why?

**Comment by francescalb** [comment]
We should have a test on this. It is important that we wuickly pick up if running sync_reasoner on EMMO does not work.

File: ontopy/ontology.py (hunk 1 above)
Comment (jesper-friis) [reply ontopy/ontology.py:947]: Add fix: `isinstance(file, str)` -> `isinstance(file, (str, Path))`

**Review by francescalb — APPROVED** [review]
As a general rule I think that untested features should be tested properly before being smerged to master.

Please make an issue that clearly decribes what needs to be tested so that it can be picked up later.

**Pull request merged**